app/models/concerns/roleable.rb

```diff
@@ -35,7 +35,6 @@ def grant_moderation!
     auto_approve_user
     enqueue_staff_welcome_message(:moderator)
     set_default_notification_levels(:moderators)
-    DiscourseEvent.trigger(:staff_granted, self, :moderator)
```

Comment:
Is this event being removed completely? We have to be careful here to maintain backwards compatibility.

Reply:
I added the event this morning as a workaround for this issue. It's safe to remove it.

Reply:
ooo icic. No worries here then.